review/src/NoMismatchedCliVersion.elm

```diff
@@ -0,0 +1,190 @@
+import Review.Rule as Rule exposing (ModuleKey, Rule)
+import String
+
+
+rule : Rule
+rule =
+    Rule.newProjectRuleSchema "NoMismatchedCliVersion" initialContext
+        |> Rule.withExtraFilesProjectVisitor staticFilesVisitor
+            [ FilePattern.include "package.json" ]
+        |> Rule.withModuleVisitor moduleVisitor
+        |> Rule.withModuleContext
+            { fromProjectToModule = fromProjectToModule
+            , fromModuleToProject = fromModuleToProject
+            , foldProjectContexts = foldProjectContexts
+            }
+        |> Rule.fromProjectRuleSchema
+
+
+initialContext : ProjectContext
+initialContext =
+    { versionFromPackageJson = Nothing
+    }
+
+
+type alias ProjectContext =
+    { versionFromPackageJson : Maybe String
+    }
+
+
+type alias ModuleContext =
+    { versionFromPackageJson : Maybe String
+    , isCliModule : Bool
+    }
+
+
+fromProjectToModule : ModuleKey -> Node ModuleName -> ProjectContext -> ModuleContext
+fromProjectToModule _ moduleName context =
+    let
+        isCliModule =
+            Node.value moduleName == [ "Cli" ]
+    in
+    { versionFromPackageJson = context.versionFromPackageJson
+    , isCliModule = isCliModule
+    }
+
+
+fromModuleToProject : ModuleKey -> Node ModuleName -> ModuleContext -> ProjectContext
+fromModuleToProject _ _ context =
+    { versionFromPackageJson = context.versionFromPackageJson
+    }
+
+
+foldProjectContexts : ProjectContext -> ProjectContext -> ProjectContext
+foldProjectContexts x y =
+    let
+        versionFromPackageJson =
+            case
+                ( x.versionFromPackageJson, y.versionFromPackageJson )
+            of
+                ( Just version, _ ) ->
+                    Just version
+
+                ( _, Just version ) ->
+                    Just version
+
+                _ ->
+                    Nothing
+    in
+    { versionFromPackageJson = versionFromPackageJson
+    }
+
+
+staticFilesVisitor :
+    Dict String { fileKey : Rule.ExtraFileKey, content : String }
+    -> ProjectContext
+    -> ( List (Rule.Error { useErrorForModule : () }), ProjectContext )
+staticFilesVisitor files context =
+    case Dict.values files of
+        packageJson :: [] ->
+            case Json.Decode.decodeString (Json.Decode.field "version" Json.Decode.string) packageJson.content of
+                Ok version ->
+                    ( []
+                    , { context
+                        | versionFromPackageJson = Just version
+                      }
+                    )
+
+                Err err ->
+                    ( [ Rule.errorForExtraFile packageJson.fileKey
+                            { message = "Failed to decode package.json"
+                            , details = [ Json.Decode.errorToString err ]
+                            }
+                            Range.empty
+                      ]
+                    , context
+                    )
+
+        _ ->
+            -- TODO: handle error
```

Comment:
This would be for the "no `package.json` found" case. I didn't think I'd be able to construct an error to report here, but looks like I could use `globalError` https://package.elm-lang.org/packages/jfmengels/elm-review/2.13.0/Review-Rule#globalError.